src/main/java/net/sf/jabref/gui/maintable/SpecialMainTableColumns.java

```diff
@@ -0,0 +1,188 @@
+            if (prio != null) {
+                return prio.createLabel();
+            } else {
+                return null;
+            }
+        }
+    };
+
+    public static final MainTableColumn READ_STATUS_COLUMN = new MainTableColumn(SpecialFieldsUtils.FIELDNAME_READ,
+            new String[] {SpecialFieldsUtils.FIELDNAME_READ},
+            new JLabel(ReadStatus.getInstance().getRepresentingIcon())) {
+
+        @Override
+        public Object getColumnValue(BibtexEntry entry) {
+
+            SpecialFieldValue status = ReadStatus.getInstance()
+                    .parse(entry.getField(SpecialFieldsUtils.FIELDNAME_READ));
+            if (status != null) {
+                return status.createLabel();
+            } else {
+                return null;
+            }
+        }
+    };
+
+    public static final MainTableColumn RELEVANCE_COLUMN = createIconColumn(SpecialFieldsUtils.FIELDNAME_RELEVANCE,
+            new String[] {SpecialFieldsUtils.FIELDNAME_RELEVANCE},
+            new JLabel(Relevance.getInstance().getRepresentingIcon()));
+
+    public static final MainTableColumn PRINTED_COLUMN = createIconColumn(SpecialFieldsUtils.FIELDNAME_PRINTED,
+            new String[] {SpecialFieldsUtils.FIELDNAME_PRINTED},
+            new JLabel(Printed.getInstance().getRepresentingIcon()));
+
+    public static final MainTableColumn QUALITY_COLUMN = createIconColumn(SpecialFieldsUtils.FIELDNAME_QUALITY,
+            new String[] {SpecialFieldsUtils.FIELDNAME_QUALITY},
+            new JLabel(Quality.getInstance().getRepresentingIcon()));
+
+
+    public static final MainTableColumn FILE_COLUMN = new MainTableColumn(Globals.FILE_FIELD,
+            new String[] {Globals.FILE_FIELD}, new JLabel(IconTheme.JabRefIcon.FILE.getSmallIcon())) {
+
+        @Override
+        public Object getColumnValue(BibtexEntry entry) {
+            // We use a FileListTableModel to parse the field content:
+            FileListTableModel fileList = new FileListTableModel();
+            fileList.setContent(entry.getField(Globals.FILE_FIELD));
+            if (fileList.getRowCount() > 1) {
+                return new JLabel(IconTheme.JabRefIcon.FILE_MULTIPLE.getSmallIcon());
+            } else if (fileList.getRowCount() == 1) {
+                ExternalFileType type = fileList.getEntry(0).getType();
+                if (type != null) {
+                    return type.getIconLabel();
+                }
+            }
+
+            return null;
+        }
+    };
+
+    /**
+     * Creates a MainTableColumn which shows an icon instead textual content
+     *
+     * @param columnName the name of the column
+     * @param fields     the entry fields which should be shown
+     * @return the crated MainTableColumn
+     */
+    public static MainTableColumn createIconColumn(String columnName, String[] fields, JLabel iconLabel) {
+        return new MainTableColumn(columnName, fields, iconLabel) {
```

Comment:
Is this method every called with more than one field? Could it be simplified to just passing one field? (I don't like arrays :)